sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/ParentBasedSampler.java

```diff
@@ -104,28 +103,20 @@ public boolean equals(Object o) {
 
     ParentBasedSampler that = (ParentBasedSampler) o;
 
-    if (!Objects.equals(root, that.root)) {
-      return false;
-    }
-    if (!Objects.equals(remoteParentSampled, that.remoteParentSampled)) {
-      return false;
-    }
-    if (!Objects.equals(remoteParentNotSampled, that.remoteParentNotSampled)) {
-      return false;
-    }
-    if (!Objects.equals(localParentSampled, that.localParentSampled)) {
-      return false;
-    }
-    return Objects.equals(localParentNotSampled, that.localParentNotSampled);
+    return root.equals(that.root)
```

Comment:
Fields are never null

Reply:
The annotations on the constructor params indicate otherwise.  If this is true, we should change that.  Maybe even `@NotNull` them?

Reply:
note that they get replaced with the defaults if null in the constructor impl.

Reply:
Yeah - it's a bit confusing, let me send another PR to replace Nullable with setting the defaults in the builder.

Reply:
Got it.